account_invoice_pricelist/model/account_invoice.py

```diff
@@ -33,26 +33,56 @@ class AccountInvoice(models.Model):
 
-    @api.multi
-    def onchange_partner_id(
-            self, type, partner_id, date_invoice=False, payment_term=False,
-            partner_bank_id=False, company_id=False):
-        partner_obj = self.env['res.partner']
-        res = super(AccountInvoice, self).onchange_partner_id(
-            type, partner_id, date_invoice=date_invoice,
-            payment_term=payment_term, partner_bank_id=partner_bank_id,
-            company_id=company_id)
+    @api.onchange('partner_id', 'company_id')
+    def _onchange_partner_id(self):
+           
+    
+#    , type, partner_id, date_invoice=False, payment_term=False,
+#            partner_bank_id=False, company_id=False):
+                
+#        partner_obj = self.env['res.partner']
+        res = super(AccountInvoice, self)._onchange_partner_id()
+#            type, partner_id, date_invoice=date_invoice,
+#            payment_term=payment_term, partner_bank_id=partner_bank_id,
+#            company_id=company_id)
         pricelist_id = False
-        if partner_id:
-            partner = partner_obj.browse(partner_id)
-            if type in ('out_invoice', 'out_refund'):
+
+        if self.partner_id:
+#            partner = partner_obj.browse(partner_id)
+            if self.type in ('out_invoice', 'out_refund'):
                 # Customer Invoices
-                pricelist_id = partner.property_product_pricelist.id
-            elif type in ('in_invoice', 'in_refund'):
+                pricelist_id = self.partner_id.property_product_pricelist
+            elif self.type in ('in_invoice', 'in_refund'):
                 # Supplier Invoices
-                if partner._model._columns.get(
+                if self.partner_id._model._columns.get(
                         'property_product_pricelist_purchase', False):
                     pricelist_id =\
-                        partner.property_product_pricelist_purchase.id
-        res['value']['pricelist_id'] = pricelist_id
+                        self.partner_id.property_product_pricelist_purchase
+            self.pricelist_id = pricelist_id
+#        res['value']['pricelist_id'] = pricelist_id
         return res
+    
+class AccountInvoiceLine(models.Model):
+    _inherit = "account.invoice.line"
+    
+    @api.onchange('product_id')
+    def _onchange_product_id(self):
+        res = super(AccountInvoiceLine, self)._onchange_product_id()
+        
+        if self.invoice_id.pricelist_id:
+            pricelist_id=self.invoice_id.pricelist_id
+            product = self.product_id.with_context(
+                                         partner=self.partner_id.id,
+                                         quantity=self.quantity,
+                                         date=self.invoice_id.date_invoice,
+                                         pricelist= pricelist_id.id,
+                                         uom=self.uom_id.id
+                                        )
+            print("price unit %s" % product.price)
```

Comment:
Do not leave print statements in PR. If desirable merge print statements into one message for _logger.debug(...)